Objects/complexobject.c

```diff
@@ -905,6 +905,74 @@ complex_subtype_from_string(PyTypeObject *type, PyObject *v)
     return result;
 }
 
+static PyObject *
+actual_complex_new(PyTypeObject *type, PyObject *args, PyObject *kwargs)
+{
+    PyObject *tmp;
+    PyObject *res = NULL;
+    PyNumberMethods *nbr;
+    int own_arg = 0;
+
+    if (PyTuple_GET_SIZE(args) > 1 || (kwargs != NULL && PyDict_GET_SIZE(kwargs))) {
+        return complex_new(type, args, kwargs);
+    }
+    if (!PyTuple_GET_SIZE(args)) {
+        return complex_subtype_from_doubles(type, 0, 0);
+    }
+
+    PyObject *arg = PyTuple_GET_ITEM(args, 0);
+    /* Special-case for a single argument when type(arg) is complex. */
+    if (PyComplex_CheckExact(arg) && type == &PyComplex_Type) {
+        /* Note that we can't know whether it's safe to return
+           a complex *subclass* instance as-is, hence the restriction
+           to exact complexes here.  If either the input or the
+           output is a complex subclass, it will be handled below
+           as a non-orthogonal vector.  */
+        return Py_NewRef(arg);
+    }
+    if (PyUnicode_Check(arg)) {
+        return complex_subtype_from_string(type, arg);
+    }
+    tmp = try_complex_special_method(arg);
+    if (tmp) {
+        arg = tmp;
+        own_arg = 1;
+    }
+    else if (PyErr_Occurred()) {
+        return NULL;
+    }
+
+    if (PyComplex_Check(arg)) {
+        /* Note that if arg is of a complex subtype, we're only
+           retaining its real & imag parts here, and the return
+           value is (properly) of the builtin complex type. */
+        Py_complex c = ((PyComplexObject*)arg)->cval;
+        res = complex_subtype_from_doubles(type, c.real, c.imag);
+    }
+    else if ((nbr = Py_TYPE(arg)->tp_as_number) != NULL &&
```

Comment:
One branch missed:
```
     952         [ +  - ]:         55 :     else if ((nbr = Py_TYPE(arg)->tp_as_number) != NULL &&
     953   [ +  +  +  + ]:         55 :              (nbr->nb_float != NULL || nbr->nb_index != NULL))
```

Reply:
Hmm, branch coverage output for C code looks cryptic.  Try ``complex([])``, i.e. nbr != NULL, but it has no nb_float/index, that works for me.

Reply:
I think `complex(MyInt())` (where `MyInt` has the `__int__` method already covers this).

Reply:
No.  Oh, I see: that should be ``nbr == NULL`` condition.  So ``complex(object())`` will work too.